hicexplorer/hicAggregateContacts.py

```diff
@@ -505,192 +709,113 @@ def main(args=None):
     ma = hm.hiCMatrix(args.matrix)
     ma.maskBins(ma.nan_bins)
     ma.matrix.data[np.isnan(ma.matrix.data)] = 0
-
-    bin_size = ma.getBinSize()
     ma.maskBins(ma.nan_bins)
     ma.matrix.data = ma.matrix.data
     new_intervals = hicexplorer.utilities.enlarge_bins(ma.cut_intervals)
     ma.setCutIntervals(new_intervals)
-    min_dist, max_dist = args.range.split(":")
 
     if args.chromosomes:
         ma.keepOnlyTheseChr(args.chromosomes)
-    chrom_sizes = ma.get_chromosome_sizes()
-    chrom_list = chrom_sizes.keys()
-    log.info("checking range {}-{}".format(min_dist, max_dist))
-    min_dist = int(min_dist)
-    max_dist = int(max_dist)
-    assert min_dist < max_dist, "Error lower range larger than upper range"
-
-    if args.transform == "z-score":
-        # use zscore matrix
-        log.info("Computing z-score matrix. This may take a while.\n")
-        ma.convert_to_zscore_matrix(maxdepth=max_dist * 2.5, perchr=True)
-    elif args.transform == "obs/exp":
-        # use zscore matrix
-        log.info("Computing observed vs. expected matrix. This may take a while.\n")
-        ma.convert_to_obs_exp_matrix(maxdepth=max_dist * 2.5, perchr=True)
 
-    min_dist_in_bins = int(min_dist) // bin_size
-    max_dist_in_bins = int(max_dist) // bin_size
+    default_range = '1000000:20000000'
+    if args.range is None:
+        if args.mode == "intra-chr":
+            log.warning("You have not set any range. This is by default set to {} for intra-chr.".format(default_range))
+        args.range = default_range
+    min_dist, max_dist = args.range.split(":")
```

Comment:
This is where I would put the tests to check the min_dist, max_dist and store them in agg_info.

Reply:
What tests do you mean?

Reply:
```
        min_dist, max_dist = range.split(":")
        log.info("checking range {}-{}".format(min_dist, max_dist))
        assert int(min_dist) < int(max_dist), "Error lower range larger than upper range"
        min_dist_in_bins = int(min_dist) // bin_size
        max_dist_in_bins = int(max_dist) // bin_size
```